src/rockstor/storageadmin/views/rockon_id.py

```diff
@@ -145,13 +145,11 @@ def post(self, request, rid, command):
                         cco.val = cc_map[c]
                         cco.save()
                     for e in env_map.keys():
-                        if (not DContainerEnv.objects.filter(
+                        if (DContainerEnv.objects.filter(
```

Comment:
@MFlyer, my pr is not hiding anything. Actually the error that was removed is actually a mistake, if we want this error to be raised anway, it should raised somewhere else. You have expaned how rockons are stored in DContainerEnv table of rockstor database. I know that, even examined database with pgadmin just to find out there is no error there an nothing is missed. But the problem I'm trying to solve is when rockon is installed in rockon_id.py.

Imagine we have some rockon Rock1. It consists of two containers: ContainerA and ContainerB. ContainerA has ENVA1 and ENVA2 variables and ContainerB has ENVB1. And that's what happens (for now):

[`env_map = request.data.get('environment', {})`](https://github.com/rockstor/rockstor-core/blob/master/src/rockstor/storageadmin/views/rockon_id.py#L94) -here we read all variables with user-assigned values in a **plain** list. In our example it would be `[['ENVA1', 'valuea1'], ['ENVA2', 'valuea2'], ['ENVB1', 'valueb1']]`
[`for co in containers:`](https://github.com/rockstor/rockstor-core/blob/master/src/rockstor/storageadmin/views/rockon_id.py#L96) - here we run through all the containers in rockon
[`--for e in env_map.keys()`](https://github.com/rockstor/rockstor-core/blob/master/src/rockstor/storageadmin/views/rockon_id.py#L147) - here we run though all environment variables in `env_map` we have filled with values before
[`----if (not DContainerEnv.objects.filter(container=co, key=e).exists()):`](https://github.com/rockstor/rockstor-core/blob/master/src/rockstor/storageadmin/views/rockon_id.py#L148) - checking if this container - variable relation is stored in database. ContainerA: ENVA1 is OK. ENVA2 is OK and ENVB1 is not OK and an error will raise. And because `env_map` contains everything this code will allways check all variables from all the containers even when it should not. Thus no multicontainer environment will ever work (with defined environment variables)

[Here is real json](https://github.com/anatox/rockon-registry/blob/nextcloud/nextcloud.json). It's not ready yet but ok to test for this error :) The install will fail when checking `nextcloud` container for `MYSQL_ROOT_PASSWORD` from `nextcloud-mariadb` container